src/lstm_cell.js

```diff
@@ -0,0 +1,145 @@
+'use strict';
+
+import {add, mul} from './binary.js';
+import {matmul} from './matmul.js';
+import {Scalar} from './lib/tensor.js';
+import {sigmoid} from './sigmoid.js';
+import {slice} from './slice.js';
+import {tanh} from './tanh.js';
+import {transpose} from './transpose.js';
+import {validateLstmCellParams} from './lib/validate-input.js';
+
+/**
+ *A single time step of the Long Short-Term Memory [LSTM] recurrent network
+ *using a cell state, an input, output, and forget gate to compute the cell
+ *state and the hidden state of the next time step that rolls into the output
+ *across the temporal sequence of the network.
+ * @param {Tensor} input
+ * @param {Tensor} weight
+ * @param {Tensor} recurrentWeight
+ * @param {Tensor} hiddenState
+ * @param {Tensor} cellState
+ * @param {Number} hiddenSize
+ * @param {MLLstmCellOptions} options
+ * @return {Tensor}
+ */
+export function lstmCell(input, weight, recurrentWeight, hiddenState, cellState, hiddenSize,
+    {bias, recurrentBias, peepholeWeight,
+      layout = 'iofg', activations = [sigmoid, tanh, tanh]}={}) {
+  validateLstmCellParams(...arguments);
+  const zero = new Scalar(0);
+  const inputSize = input.shape[1];
+  const starts = layout === 'iofg' ? {i: 0, o: hiddenSize, f: 2 * hiddenSize, g: 3 *hiddenSize} :
+  {i: 0, f: hiddenSize, g: 2 * hiddenSize, o: 3 * hiddenSize};
```

Comment:
Could be easier for future readers to visual parse if the indentation was aligned:

```suggestion
  const starts = (layout === 'iofg') ? {i: 0, o: hiddenSize, f: 2 * hiddenSize, g: 3 * hiddenSize} :
                                       {i: 0, f: hiddenSize, g: 2 * hiddenSize, o: 3 * hiddenSize};